app/models/like.rb

```diff
@@ -0,0 +1,35 @@
+class Like < ActiveRecord::Base
+  validates :user_id, :likable_id, :likable_type, presence: true
+  validates :user_id, uniqueness: { scope: [:likable_id, :likable_type] }
+
+  belongs_to :user
+  belongs_to :likable, :polymorphic => true
+
+  
+  def self.display_likes_msg(likable)
+    return nil if likable.likes.empty?
+    creator_name(likable.likes) + like_msg_base(likable.likes)      
+  end
+
+  def self.creator_name(like_list)
+    str = like_list.inject("") do |str, like|
+      str + "#{like.user.profile.full_name} and "
+    end
+    str[0..-5]
```

Comment:
If these methods are being used for the view, perhaps they could be placed in a helper